sormas-backend/src/main/java/de/symeda/sormas/backend/report/AggregateReportFacadeEjb.java

```diff
@@ -185,42 +183,147 @@ public List<AggregatedCaseCountDto> getIndexList(AggregateReportCriteria criteri
+			}
+			if (groupingLevel.equals(AggregateReportGroupingLevel.POINT_OF_ENTRY)) {
+				filter = CriteriaBuilderHelper.and(cb, filter, cb.or(cb.isNotNull(root.get(AggregateReport.POINT_OF_ENTRY))));
+			}
+
 				Join<AggregateReport, Region> regionJoin = root.join(AggregateReport.REGION, JoinType.LEFT);
 				List<Path<Object>> regionPath = Arrays.asList(regionJoin.get(Region.NAME), regionJoin.get(Region.ID));
 				expressions.addAll(regionPath);
 				selectionList.addAll(regionPath);
-			}
 
-			if (shouldIncludeDistrict(groupingLevel)) {
 				Join<AggregateReport, District> districtJoin = root.join(AggregateReport.DISTRICT, JoinType.LEFT);
 				List<Path<Object>> districtPath = Arrays.asList(districtJoin.get(District.NAME), districtJoin.get(District.ID));
 				expressions.addAll(districtPath);
 				selectionList.addAll(districtPath);
-			}
 
-			if (shouldIncludeHealthFacility(groupingLevel)) {
 				Join<AggregateReport, Facility> facilityJoin = root.join(AggregateReport.HEALTH_FACILITY, JoinType.LEFT);
 				List<Path<Object>> facilityPath = Arrays.asList(facilityJoin.get(Facility.NAME), facilityJoin.get(Facility.ID));
 				expressions.addAll(facilityPath);
 				selectionList.addAll(facilityPath);
-			}
 
-			if (shouldIncludePointOfEntry(groupingLevel)) {
 				Join<AggregateReport, PointOfEntry> pointOfEntryJoin = root.join(AggregateReport.POINT_OF_ENTRY, JoinType.LEFT);
-				List<Path<Object>> pointOfEntryPath = Arrays.asList(pointOfEntryJoin.get(PointOfEntry.ID), pointOfEntryJoin.get(PointOfEntry.NAME));
+				List<Path<Object>> pointOfEntryPath = Arrays.asList(pointOfEntryJoin.get(PointOfEntry.NAME), pointOfEntryJoin.get(PointOfEntry.ID));
 				expressions.addAll(pointOfEntryPath);
 				selectionList.addAll(pointOfEntryPath);
-			}
 		}
 
+		selectionList.addAll(
+			Arrays.asList(
+				joins.getReportingUser().get(User.UUID),
+				joins.getReportingUser().get(User.FIRST_NAME),
+				joins.getReportingUser().get(User.LAST_NAME),
+				root.get(AggregateReport.CHANGE_DATE)));
+		expressions.addAll(
+			Arrays.asList(
+				joins.getReportingUser().get(User.UUID),
+				joins.getReportingUser().get(User.FIRST_NAME),
+				joins.getReportingUser().get(User.LAST_NAME),
+				root.get(AggregateReport.CHANGE_DATE)));
+
 		cq.multiselect(selectionList);
 
+		if (filter != null) {
+			cq.where(filter);
+		}
+
 		cq.groupBy(expressions);
 
-		List<AggregatedCaseCountDto> resultList = em.createQuery(cq).getResultList();
+		List<AggregatedCaseCountDto> queryResult = em.createQuery(cq).getResultList();
 		Map<Disease, AggregatedCaseCountDto> reportSet = new HashMap<>();
 
+		List<AggregatedCaseCountDto> resultList = new ArrayList<>();
+		Map<AggregatedCaseCountDto, List<AggregatedCaseCountDto>> subJurisdictionsToBeMerged = new HashMap<>();
+
+		final AggregateReportGroupingLevel finalGroupingLevel = groupingLevel;
+		for (AggregatedCaseCountDto aggregatedCaseCountDto : queryResult) {
```

Comment:
I'd like you to add comments to this whole section because it's really tough to read. Also, please consider using stream methods like `removeIf` and potentially even `reduce` if possible in order to shorten the code and make it more readable.